crates/nargo/src/cli/mod.rs

```diff
@@ -143,7 +155,16 @@ fn write_inputs_to_file<P: AsRef<Path>>(
         dir_path
     };
 
-    let serialized_output = format.serialize(w_map)?;
+    // If it exists, insert return value into input map so it's written to file.
+    let serialized_output = match return_value {
+        Some(return_value) => {
+            let mut input_map = input_map.clone();
+            input_map.insert(MAIN_RETURN_NAME.to_owned(), return_value.clone());
+            format.serialize(&input_map)?
+        }
+        None => format.serialize(input_map)?,
```

Comment:
Can we add a comment here explaining why we insert the MAIN_RETURN_NAME parameter on a clone of the map instead of in the input map itself? The effect is that we do not pass it to `verify_proof` if the checked flag is set in prove_cmd.rs, we should elaborate on this I think.